app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -2,30 +2,39 @@ module ManageIQ::Providers
   class Hawkular::MiddlewareManager::LiveMetricsCapture
     class MetricValidationError < RuntimeError; end
 
-    MetricsResource = Struct.new(:id, :feed, :path)
-
     def initialize(target)
       @target = target
       @ems = @target.ext_management_system
       @gauges = @ems.metrics_client.gauges
       @counters = @ems.metrics_client.counters
       @avail = @ems.metrics_client.avail
+      @included_children = @target.class.included_children
+      @supported_metrics = @target.class.supported_metrics
     end
 
-    def metrics_available
-      resource = MetricsResource.new
-      resource.id = @target.nativeid
-      resource.feed = extract_feed(@target.ems_ref)
-      resource.path = @target.ems_ref
-      @ems.metrics_resource(resource.path).collect do |metric|
+    def fetch_metrics_available
+      metrics_available = @ems.metrics_resource(@target.ems_ref).collect do |metric|
         next unless @target.class.supported_metrics[metric.name]
```

Comment:
based on [this assignment](https://github.com/ManageIQ/manageiq/pull/9134/files#diff-70795fbbcc6754e2afd15dd154dbc330R12), looks like makes sense to use the supported_metrics variable here?